### ISSUE TITLE
Client downloads flags and maps from server if they don't have them

### ISSUE THREAD

**Issue by andy5995** [netpanzer/netpanzer#147]
Using libcurl, or a new mod system ? We shall see.

This would fix #146

**Comment by winrid** [comment]
libcurl would be a nice stop gap solution.

**Comment by winrid** [comment]
honestly libcurl (with openssl?) would be the way to download the mods in the future, too.

**Comment by andy5995** [comment]
[mod.io](https://mod.io/) seems pretty popular. It has an API. 0ad's in-game mod downloader uses it. Maybe in combination with libcurl, I haven't looked.

Any registered user can add mods, maps, tilesets, etc. They don't show up in the in-game mod center though unless signed by a 0ad project admin. If they're not signed, they still show up on the site and can be downloaded by anyone.

**Comment by winrid** [comment]
Interesting. I was just going to have the server owner put the pak files they want on their server, and then client will download them on connect when needed. I thought this was a nice simple approach, but open to understanding the advantages of mod.io. They could always upload user-created mods to moddb.

**Comment by andy5995** [comment]
> Interesting. I was just going to have the server owner put the pak files they want on their server, and then client will download them on connect when needed.

Yes, I agree. That sounds like how @MegaGlest does it. If a player is hosting a game and a client joins, the server will offer the map automatically. The client user gets a y/n prompt to confirm before download.

If the client already has the map, then the server does a CRC check to make sure it matches.

(One thing missing from MG is an offer to "update" the map if the CRC doesn't match)

>I thought this was a nice simple approach, but open to understanding the advantages of mod.io. They could always upload user-created mods to moddb.

I accidentally went off-topic. That's a discussion for when NP has more mod capability, correct? Though you might want to register NP on mod.io in case you want to take advantage of it later.

**Comment by winrid** [comment]
> That sounds like how https://github.com/MegaGlest does it.

I got the idea from Enemy Territory.

> If the client already has the map, then the server does a CRC check to make sure it matches.

Yes, we can support signing (I think this is configurable in ET). Although thinking about it, doesn't that require

> I accidentally went off-topic. That's a discussion for when NP has more mod capability, correct? Though you might want to register NP on mod.io in case you want to take advantage of it later.

I suppose, yes, can be discussed later!
